modules/nf-core/trust4/main.nf

```diff
@@ -34,12 +36,41 @@ process TRUST4 {
     // separate forward from reverse pairs
     def (forward, reverse) = reads.collate(2).transpose()
     def paired_end_mode = reads && (meta.single_end == false) ? "-1 ${forward[0]} -2 ${reverse[0]}" : ''
+    // read format is optional
+    def readFormat = params.read_format ? "--readFormat ${params.read_format}" : ''
+    // add barcode information if present
+    def barcode = ''
+    if (meta.barcode_read) {
```

Comment:
Modules cannot rely on the `meta` map containing any information other than `id` and `single_end` (cf. [modules docs](https://nf-co.re/docs/guidelines/components/modules#types-of-meta-fields)). You should refactor this to pass the information in another way. I know this might be trickier than usual (as the docs show no pre-processing on the variable), but there are solutions. My suggestion is to consider passing `barcode_read` and `umi_read`, and leave it to pipeline maintainers to correctly fetch the information form their `meta`. [This config](https://github.com/nf-core/modules/blob/71bc6bbffc7ad08e05c4d449ca3ba0f86b000125/subworkflows/nf-core/bam_split_by_region/nextflow.config) demonstrates how it might work inside a workflow.